catalog/dags/common/elasticsearch.py

```diff
@@ -78,6 +78,29 @@ def remove_excluded_index_settings(index_config):
     return index_config
 
 
+@task
+def get_record_count_group_by_sources(es_host: str, index: str):
+    """
+    Airflow task that returns a dict where the keys are the sources, and the values are the counts.
+    The function calls Elasticsearch to run an aggs query to do a count grouped by the field "source", and parses the result into a dict.
+    """
+    es_hook = ElasticsearchPythonHook(hosts=[es_host])
+    body = {"aggs": {"unique_sources": {"terms": {"field": "source"}}}}
+
+    # Unfornately the ElasticsearchPythonHook's search function doesn't work with aggs, because it returns the "hits" key only
+    # See source code at https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/_modules/airflow/providers/elasticsearch/hooks/elasticsearch.html#ElasticsearchPythonHook
+    # Therefore using get_conn to call the search from the ES client
+    es_client = es_hook.get_conn
+    # es_result object looks like: [{'key': 'flickr', 'doc_count': 2500}, {'key': 'stocksnap', 'doc_count': 2500}]
+    es_result = es_client.search(body=body, index=index)
```

Comment:
(RE: missing index at the start)
```suggestion
    try:
        es_result = es_client.search(body=body, index=index)
    except NotFoundError as err:
        logging.warning(f"Index {index} does not appear to exist: {err}")
        return {}
```